proposals/004-skips/README.md

```diff
@@ -0,0 +1,179 @@
+    }
+}
+```
+
+See the full diff [here](https://github.com/spinkube/spin-operator/compare/main...Mossaka:spin-operator-msk:sidecars)
+
+[sidecar containers]: (https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)
+[InitContainers]: (https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+### Ports
+
+Sidecar containers must avoid using port 80, which is reserved for the Spin application. Other ports like 8080 or 8081 can be used.
+
+### Services
+
+Sidecar containers can communicate internally using `localhost`, but if external access is needed, users must create a separate service for the sidecar container.
+
+### Restrictions
+
+* Sidecar containers cannot use the same image or name as the Spin App. 
+* Sidecar containers must not use port 80.
+
+### Sample SpinApp yaml
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: simple-spinapp
+spec:
+  image: "ttl.sh/spinapp:44h"
+  replicas: 1
+  executor: containerd-shim-spin
+  volumes:
+    - name: example-volume
+      persistentVolumeClaim:
+        claimName: example-pv-claim
+  volumeMounts:
+    - name: example-volume
+      mountPath: "/mnt/data"
+  initContainers:
+    - name: logshipper
+      image: alpine:latest
+      restartPolicy: Always
+      command: ['sh', '-c', 'tail -F /opt/logs.txt']
+      volumeMounts:
+      - name: example-volume
+        mountPath: "/mnt/data"
+```
+
+Note: The `InitContainers` field is intentionally designed to follow the the design of sidecar containers in Kubernetes Pods. You will need to have `SidecarContainers` feature gate enabled in your Kubernetes cluster to use this feature. See more details [here](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/).
+
+### Init Containers
+
+Since the sidecar containers are just a special case of InitContainers, I realized that the scope of this SKIP is naturally expanded to include the support of init containers in the Spin Operator. 
+
+Examples of init containers include:
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: simple-spinapp
+spec:
+  image: "ttl.sh/spinapp:44h"
+  replicas: 1
+  executor: containerd-shim-spin
+  volumes:
+    - name: example-volume
+      persistentVolumeClaim:
+        claimName: example-pv-claim
+  volumeMounts:
+    - name: example-volume
+      mountPath: "/mnt/data"
+  initContainers:
+  - name: init-myservice
+    image: busybox:1.28
+    command: ['sh', '-c', "until nslookup myservice.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done"]
+  - name: init-mydb
+    image: busybox:1.28
+    command: ['sh', '-c', "until nslookup mydb.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for mydb; sleep 2; done"]
+```
+
+## Alternatives Considered
+
+1. Use `Containers` instead of `InitContainers` in the SpinApp CRD.
```

Comment:
could you add a little detail on why these ids are were considered and then discarded? (pros/cons)